graphbolt/src/concurrent_id_hash_map.cc

```diff
@@ -199,6 +209,30 @@ inline void ConcurrentIdHashMap<IdType>::InsertAndSet(IdType id, IdType value) {
   hash_map_.data_ptr<IdType>()[getValueIndex(pos)] = value;
 }
 
+template <typename IdType>
+void ConcurrentIdHashMap<IdType>::InsertAndSetSmaller(IdType id, IdType value) {
+  IdType pos = (id & mask_), delta = 1;
+  IdType* hash_map_data = hash_map_.data_ptr<IdType>();
+  InsertState state = AttemptInsertAt(pos, id);
+  while (state == InsertState::OCCUPIED) {
+    Next(&pos, &delta);
+    state = AttemptInsertAt(pos, id);
+  }
+
+  if (state == InsertState::INSERTED) {
+    hash_map_data[getValueIndex(pos)] = value;
```

Comment:
There is a data race here. We need to perform an atomic store operation here.

Reply:
Actually, all store operations need to be atomicMax operations so this if statement I think should be removed.

Reply:
Instead, we need to ensure that `empty_key == std::numeric_limits<IdType>::max()`;